src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -1053,16 +1055,71 @@ public void Connect(IPAddress[] addresses, int port)
                 throw new NotSupportedException(SR.net_invalidversion);
             }
 
+// Disable CS0162: Unreachable code detected
+//
+// SuportsMultipleConnectAttempts is a constant; when false, the following lines will trigger CS0162.
+#pragma warning disable 162
             Exception lastex = null;
-            foreach (IPAddress address in addresses)
+            if (SocketPal.SupportsMultipleConnectAttempts)
             {
-                if (CanTryAddressFamily(address.AddressFamily))
+                foreach (IPAddress address in addresses)
+                {
+                    if (CanTryAddressFamily(address.AddressFamily))
+                    {
+                        try
+                        {
+                            Connect(new IPEndPoint(address, port));
+                            lastex = null;
+                            break;
+                        }
+                        catch (Exception ex)
+                        {
+                            if (ExceptionCheck.IsFatal(ex)) throw;
+                            lastex = ex;
+                        }
+                    }
+                }
+            }
+            else
+            {
+                EndPoint endpoint = null;
```

Comment:
Actually, I was talking about disk footprint. Decreasing it was the primary driver for CoreFX, at least in the initial discussions.